### PR TITLE
feat(network-simple): use network-simple instead of manually handling sockets

### PR DESCRIPTION
**Pull request by mdibaiee**

### CODE REVIEW

**Comment by amir-sabbaghi** [comment]
I don't merge two feature in one pull request. Also SSL should be optional. Someone may want to start a plain HTTP proxy server.

**Comment by mdibaiee** [comment]
@amir-sabbaghi okay

File: proxy.cabal [hunk 1]
```diff
@@ -11,20 +11,28 @@ author: Amir Sabbaghi
 data-dir: ""
 
 executable proxy
-    build-depends: base == 4.8.*, network >=2.6 && <2.7, proxy
+    build-depends: base < 4.8, network >=2.6 && <2.7, proxy
```
Comment (amir-sabbaghi) [inline proxy.cabal:14]: What is the problem with base 4.8?

Comment (mdibaiee) [reply proxy.cabal:14]: `base 4.8` is for GHC `7.10.1`, but we're using `7.8.4`.

https://wiki.haskell.org/Base_package

**Pull request merged**